src/Stratis.Bitcoin.Features.BlockStore/IPrunedBlockRepository.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Threading.Tasks;
+using NBitcoin;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.BlockStore
+{
+    /// <summary>
+    /// Prunes the block store database by deleting blocks lower than a certain height.
+    /// </summary>
+    public interface IPrunedBlockRepository
+    {
+        /// <summary>
+        /// Initializes the pruned block repository.
+        /// </summary>
+        /// <returns>The awaited task.</returns>
+        Task InitializeAsync();
+
+        /// <summary>
+        /// Compacts the block and transaction database by resaving the database file without
+        /// all the deleted references.
+        /// </summary>
+        /// <param name="consensusTip">The current tip of consensus.</param>
+        /// <param name="network">The network the node is running on.</param>
+        /// <param name="nodeInitializing">Indicates whether or not this method is called from node startup or not.</param>
+        /// <returns>The awaited task.</returns>
+        Task PruneDatabase(ChainedHeader consensusTip, Network network, bool nodeInitializing);
```

Comment:
This is not pruning, it's a cleanup. 

Please change it to remove the blocks during the runtime instead of removing them on startup and shutdown. 

Reasons why cleanup is not good enough: 

1) It makes startup and shutdown a lot longer
2) Node still can store whole blockchain if you sync it in IBD. It doesn't work for devices with small amount of HDD. Imagine you want to sync BTC network and you have 20 gigs on your hard drive. Your only way would be to stop and start the node every 10-15 gb or smth.

Reply:
This is pruning? We are deleting blocks from the database whilst keeping a certain amount of blocks on disk?

It is already pruning during run time., see the `LightWalletBlockStoreService` which runs a period task to prune the block store.

Compacting only releases unused disk space on start up and shut down.

The only way start up will take long is if you enable pruning on a full synced node. We can change this so that pruning is only allowed on a empty data folder node?